File: frontend/src/app/utils/format.utils.ts
export class FormatUtils {
	public static formatTelephone(telephone: string | number): string {
		const tel = telephone.toString();
		return `(${ tel.substring(0, 2) }) ${ tel.substring(2, 7) }-${ tel.substring(7) }`;
	}

	public static formatCurrency(value: number): string {
		return Intl.NumberFormat('pt-BR', { style: 'currency', currency: 'BRL' }).format(value);
	}
}

File: backend/src/main/java/br/com/chadschoperia/utils/FormatUtils.java
package br.com.chadschoperia.utils;

import lombok.experimental.UtilityClass;

@UtilityClass
public class FormatUtils {
	public static String formatTelephone(String tel) {
		return String.format("(%s) %s-%s", tel.substring(0, 2), tel.substring(2, 7), tel.substring(7));
	}

}
